os/test/src/PathTests.scala

```diff
@@ -2,16 +2,90 @@ package test.os
 
 import java.nio.file.Paths
 import java.io.File
-
 import os._
-import os.Path.{driveRoot}
+import os.Path.driveRoot
+import os.PathChunk.segmentsFromString
 import utest.{assert => _, _}
+
 import java.net.URI
 object PathTests extends TestSuite {
+  private def nonValidPathSegment(chars: String) = s"[$chars] is not a valid path segment."
```

Comment:
I think we can do a more use-focused error message here, something like 

> literal path sequence <entire-literal> used in OS-Lib cannot have redundant slashes, please use <literal-without-leading-slashes> instead

Reply:
@lihaoyi
I have provided a `NonCanonicalLiteral` exception for this case
actually there is also an edge case when sanitizing literal results in empty path, for example `//`, `/.`, `.` literals , in such cases i throw:
```scala
InvalidSegment(
      literal,
      s"Literal path sequence [$literal] doesn't affect path being formed, please remove it"
    )
``` 
If you have any better suggestions for error messages I can change them.